### PR TITLE
18023 incorrect name in breadcrumb

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#18023

*Description of changes:*
Created a separate breadcrumb for logged in users

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -225,9 +227,12 @@ export default class App extends Mixins(
...
     // Set base crumbs based on user role
     // Staff don't want the home landing page and they can't access the Manage Business Dashboard
-    if (this.isRoleStaff) {
-      // If staff, set StaffDashboard as home crumb
-      crumbs.unshift(getStaffDashboardBreadcrumb())
+    if (this.isAuthenticated) {
+      if (this.isRoleStaff) {
+        crumbs.unshift(getStaffDashboardBreadcrumb())
+      } else {
+        crumbs.unshift(getUserDashboardBreadcrumb())
```
Comment (severinbeauvais) [inline src/App.vue:234]: I think you don't need to do this.

If you're not logged in as staff, the breadcrumb link takes you to this page, which is titled "BC Registries Dashboard". It's the same page as now, but the text is wrong.

File: src/App.vue [hunk 2]
```diff
@@ -225,9 +227,12 @@ export default class App extends Mixins(
...
+      }
     } else {
       // For non-staff, set Home crumb
```
Comment (severinbeauvais) [inline src/App.vue:237]: "For non-staff, or not logged in, set Registry Dashboard Breadcrumb"

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-725-eeac624z.web.app

**Comment by janisrogers** [comment]
The changes requested in the ticket look great. I did however notice that the non-logged in user is taken to the login page, instead of the BCROS home page (https://dev.bcregistry.gov.bc.ca/) Can that be updated?

**Comment by severinbeauvais** [comment]
> The changes requested in the ticket look great. I did however notice that the non-logged in user is taken to the login page, instead of the BCROS home page (https://dev.bcregistry.gov.bc.ca/) Can that be updated?

Is it correct to tell the user in the breadcrumb that they will go to the login page? After login, they go to the BCROS dashboard (same as logged in).

**Comment by severinbeauvais** [comment]
> The changes requested in the ticket look great. I did however notice that the non-logged in user is taken to the login page, instead of the BCROS home page (https://dev.bcregistry.gov.bc.ca/) Can that be updated?

@janisrogers Is it correct to tell the user in the breadcrumb that they will go to the login page? After login, they go to the BCROS dashboard (same as logged in).

**Comment by janisrogers** [comment]
There is a non logged in version of the BCROS dashboard. I don't think we should direct the user to the login page if they haven't chosen to login.

**Comment by severinbeauvais** [comment]
> There is a non logged in version of the BCROS ~dashboard~ Main page. This is where the breadcrumb should take them. I don't think we should direct the user to the login page if they haven't chosen to login.

That's a good point / idea. Janis, let us know whether to proceed with that.

James, when Janis has approved this requirement, the non-logged-in breadcrumb should go to `registryHomeUrl` and its text should be "BC Registries and Online Services". The logged-in breadcrumb should go to `${registryHomeUrl}dashboard/${getParams()}` and its text should be "BC Registries Dashboard".

**Comment by janisrogers** [comment]
Yes, please proceed with that

File: src/App.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/App.vue:237]: Please update comment. And please rename the variable to `getRegistryHomeBreadcrumb`.

File: src/resources/breadcrumb-resources.ts [hunk 3]
```diff
@@ -10,10 +10,16 @@ export function getRegistryDashboardBreadcrumb (): BreadcrumbIF {
...
+  }
+}
+export function getUserDashboardBreadcrumb (): BreadcrumbIF {
+  const registryHomeUrl = sessionStorage.getItem('REGISTRY_HOME_URL')
+  return {
+    text: 'BC Registries Dashboard',
     href: `${registryHomeUrl}dashboard/${getParams()}`
   }
 }
-
```
Comment (severinbeauvais) [inline src/resources/breadcrumb-resources.ts:16]: Please restore blank line to keep this style consistent.

Comment (severinbeauvais) [reply src/resources/breadcrumb-resources.ts:16]: Also add blank line above line 16.

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue:234]: The requirement changed. This conversation is obsolete.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-725-eeac624z.web.app

**Comment by severinbeauvais** [comment]
James, there might be a problem with how the code determines whether or not the user is authenticated.

I tested this change in the preview build above (temp URL), and it looks fine for logged out, but after logged in, the breadcrumb is incorrect until I refresh the page again. Try it yourself. (I have an idea how to fix it.)

**Comment by severinbeauvais** [comment]
> I have an idea how to fix it.

I think the error is that `isAuthenticated` checks for the presence of the keycloak token before it's populated. Because it's a getter, its value is cached until something it depends on changes (ie, responsive), but it doesn't recognize changes to the session storage so it keeps its original False value.

Ref: https://stackoverflow.com/questions/69789213/any-way-to-make-sessionstorage-reactive-in-vue3

**Review by severinbeauvais — APPROVED** [review]
This is good enough for now but I might pick away at the code a bit tomorrow to clean something up.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-725-eeac624z.web.app

**Comment by severinbeauvais** [comment]
> This is good enough for now but I might pick away at the code a bit tomorrow to clean something up.

There was actually a bug in this (wrong getter), but I want to keep THIS change separate from the login detection change (*), so I fixed the bug using `isAuthenticated` (as part of THIS ticket) and created a new ticket for the login detection.

(*) I want us to fix the login detection properly, which is out of scope for THIS ticket.

**Pull request merged**